lib/smack/SmackRep.cpp

```diff
@@ -658,10 +658,48 @@ const Expr* SmackRep::lit(const llvm::Value* v, bool isUnsigned) {
         llvm_unreachable("Unsupported floating-point type.");
       }
       const APInt API = APF.bitcastToAPInt();
-      const APInt n_sign = API.trunc(expSize+sigSize-1);
-      const APInt sig = n_sign.trunc(sigSize-1);
-      const APInt exp = n_sign.lshr(sigSize-1);
-      return Expr::lit(APF.isNegative(), sig.toString(10, false), exp.toString(10, false), sigSize, expSize);
+      const APInt exp = API.lshr(sigSize-1).trunc(expSize);
+      APInt sig = API.trunc(sigSize-1).zext(sigSize);
+
+      if (exp.isAllOnesValue()) {
+        if (sig != 0) {
+          return Expr::lit("NaN", sigSize, expSize);
+        }
+        if (API.isNegative()) {
+          return Expr::lit("-oo", sigSize, expSize);
+        }
+        return Expr::lit("+oo", sigSize, expSize);
+      }
+
+      if (exp != 0) {
+        sig.setBit(sigSize - 1); // hidden bit
+      }
+
+      APInt bias = APInt::getSignedMaxValue(expSize);
+      if (exp == 0) {
+        --bias;
+      }
+
+      bool overflow;
+      APInt moveDec = exp.usub_ov(bias, overflow);
+      moveDec &= APInt(expSize, 3);
+
+      APInt finalExp = exp.usub_ov(bias, overflow).usub_ov(moveDec, overflow)
+        .ashr(2);
+
+      int leftSize = 4 * (*moveDec.getRawData() / 4 + 1);
```

Comment:
Do `moveDec->getRawData()` here, instead of `*moveDec.getRawData()`.

Reply:
In fact, since `moveDec->getRawData()` is used 3 times, introduce a local variable for it.

Reply:
`*moveDec.getRawData()` is equivalent to `*(moveDec.getRawData())`, not `(*moveDec).getRawData()`, which is what `moveDec->getRawData()` is equivalent to.

Reply:
Got it, my mistake.